server/runtime/src/main/java/org/infinispan/server/tasks/ServerTaskEngine.java

```diff
@@ -81,7 +82,10 @@ private <T> CompletableFuture<T> invokeTask(TaskContext context, ServerTaskWrapp
       launderParameters(context);
       MediaType requestMediaType = context.getCache().map(c -> c.getAdvancedCache().getValueDataConversion().getRequestMediaType()).orElse(MediaType.MATCH_ALL);
       context.getCache().ifPresent(c -> context.cache(c.getAdvancedCache().withMediaType(APPLICATION_OBJECT, APPLICATION_OBJECT)));
-      return runner.execute(task.getName(), context).thenApply(r -> (T) scriptConversions.convertToRequestType(r, APPLICATION_OBJECT, requestMediaType));
+      return bm.supplyBlocking(() -> runner.execute(task, context), "task-execution-" + task.getName())
```

Comment:
I would pass the `blockingManager` as an argument of `execute`. The `LocalServerTaskRunner` is the only implement that should use it.

Reply:
Updated. The local task extracts the blocking manager from the cache manager. Similar to how the server extracts the cluster executor.